src/common/login/PostLoginActions.ts

```diff
@@ -115,8 +116,51 @@ export class PostLoginActions implements PostLoginAction {
 		this.showSetupWizardIfNeeded()
 	}
 
+	// Runs the user approval check after the user has been updated or after a timeout
+	private checkApprovalAfterSync(): Promise<void> {
+		return new Promise<void>((resolve) => {
```

Comment:
I think what it's doing is really nice. Good job puling it off and investing effort into it.

I would structure it a bit differently perhaps:

1. Always add a listener
2. Start listening until we get the customer update (this will give us one promise)
3. start `delay(2000)` (this will give us the second promise) 
4. Do a Promise.race() between them which will `removeEntityListener()` and `checkApprovalStatus()`

Reply:
Thanks for the review. That is a better way to structure it. I have pushed an implementation now.